src/Hosting/DnsRecord.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * This file is part of the Acquia Platform: Cloud Data Model.
+ *
+ * (c) Acquia, Inc.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Acquia\Platform\Cloud\Hosting;
+
+use InvalidArgumentException;
+
+/**
+ * Class DnsRecord.
+ */
+class DnsRecord implements DnsRecordInterface
+{
+    /**
+     * The type of DNS record (A, AAAA, CNAME).
+     *
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * The value of the DNS record (IPv4 address, IPv6 address, CNAME).
+     *
+     * @var string
+     */
+    protected $value;
+
+    /**
+     * DnsRecord constructor.
+     *
+     * @param string $type
+     *   The type of DNS record (A, AAAA, CNAME).
+     * @param string $value
+     *   The value of the DNS record (IPv4 address, IPv6 address, CNAME).
+     */
+    public function __construct(string $type, string $value)
+    {
+        // Validate record type.
+        $allowedTypes = ['A', 'AAAA', 'CNAME'];
+        if (!in_array($type, $allowedTypes)) {
+            throw new InvalidArgumentException(sprintf(
+                '$type must be one of %s; %s provided.',
+                implode(', ', $allowedTypes),
+                $type
+            ));
+        }
+        $this->type = $type;
+
+        // Validate value based on record type.
+        if ($type == 'A' && !filter_var($value, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4)) {
+            throw new InvalidArgumentException('$value must be a valid IPv4 address.');
+        } elseif ($type == 'AAAA' && !filter_var($value, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6)) {
+            throw new InvalidArgumentException('$value must be a valid IPv6 address.');
+        } elseif ($type == 'CNAME' && !filter_var($value, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME)) {
```

Comment:
These can all be `if` since they're `throw`ing exceptions. Also, what do you think about moving this to a new validation method?

Reply:
Changed these to `if`s. There's not much going on in this constructor (and the class is so simple, there likely never will be) so I'm not sure breaking it out will improve readability.